components/camel-aws/camel-aws2-s3/src/main/java/org/apache/camel/component/aws2/s3/stream/AWS2S3StreamUploadProducer.java

```diff
@@ -244,29 +262,41 @@ private CompleteMultipartUploadResponse completeUpload(UploadState state) {
                         .uploadId(state.initResponse.uploadId())
                         .build();
 
-        CompleteMultipartUploadResponse uploadResult = getEndpoint().getS3Client().completeMultipartUpload(compRequest);
+        try {
+            final CompleteMultipartUploadResponse uploadResult
+                    = getEndpoint().getS3Client().completeMultipartUpload(compRequest);
 
-        // Converting the index to String can cause extra overhead
-        if (LOG.isInfoEnabled()) {
-            LOG.info("Completed upload for the part {} with etag {} at index {}", part, uploadResult.eTag(),
-                    state.index);
+            // Converting the index to String can cause extra overhead
+            if (LOG.isInfoEnabled()) {
+                LOG.info("Completed upload for the part {}, multipart {} with etag {} at index {}", part, state.multipartIndex,
+                        uploadResult.eTag(),
+                        state.index);
+            }
+            part.getAndIncrement();
+            return uploadResult;
+        } catch (Exception e) {
+            LOG.error("Error completing multipart updload");
```

Comment:
Change to WARN and we likely need to include the caused exception so the user has a chance to understand why it failed. Also adjust the message to say that the upload will be aborted.